filesystem.md

```diff
@@ -106,3 +107,28 @@ The `Storage` facade may be used to interact with any of your configured disks.
 #### Delete A Directory
 
 	Storage::deleteDirectory($directory);
+
+<a name="custom-filesystems"></a>
+## Custom Filesystems
+If you need to use a filesystem that is not provided out of the box, you may configure one using a service provider. Visit [Flysystem](https://github.com/thephpleague/flysystem#adapters) to view a list of available options.
+
+In order to setup your custom filesystem you call the `extend` method on the `Illuminate\Filesystem\FilesystemManager` class. This class is available to you through the `Illuminate\Contracts\Filesystem\Factory` interface which can be injected into the `boot` method in your service provider.
```

Comment:
Jumping straight into the abstract definition here is hard to follow for someone who hasn't done it.

I'd suggest being a little more concrete, or doing the example first? Maybe something like:

```
In order to set up your custom file system, you'll need to create Service a Provider (for example, `DropboxFilesystemServiceProvider`). In its boot method, you'll want to inject an instance of the `Illuminate\Contracts\Filesystem\Factory` contract, and then you'll call the `extend` method of the injected instance.`
```

... etc. etc.